bach/bach/describe.py

```diff
@@ -74,21 +55,72 @@ def __init__(
         if not self.df.data:
             raise ValueError('Cannot describe a Dataframe without columns')
 
-        self.include = _get_casted_filtering_dtypes(include)
-        self.exclude = _get_casted_filtering_dtypes(exclude)
+        self.series_to_describe = self.determine_series_to_describe(self.df, include, exclude)
 
         self.datetime_is_numeric = datetime_is_numeric
         self.percentiles = percentiles or [0.25, 0.5, 0.75]
 
         if self.percentiles and any(pt < 0 or pt > 1 for pt in self.percentiles):
             raise ValueError('percentiles should be between 0 and 1.')
 
+    @staticmethod
+    def determine_series_to_describe(
+            df: DataFrame,
+            include: Optional[Union[str, Sequence[str]]],
+            exclude: Optional[Sequence[str]]
+    ) -> List[str]:
+        """
+        Process the include and exclude parameters and determine which series of the dataframe should be
+        described.
+        Validates that include and exclude don't overlap, if they are both explicitly set.
+        :return: The names of the series from the dataframe to describe
+        """
+        include_dtypes: Set[str]
+        exclude_dtypes: Set[str]
+
+        numeric_dtypes = {col.dtype for col in df.data.values() if isinstance(col, SeriesAbstractNumeric)}
+        all_dtypes = {col.dtype for col in df.data.values()}
+
+        # process `include` parameter
+        if include is None:
+            if numeric_dtypes:
+                include_dtypes = numeric_dtypes
+            else:
+                include_dtypes = all_dtypes
```

Comment:
Let's try to avoid nested `if-else`
```suggestion
            include_dtypes = numeric_dtypes or all_dtypes
```

Reply:
OK.

I think both is equally valid tho. `x or y` is still an if-else in my book ;)